src/main/java/seedu/partyplanet/logic/commands/DeleteContactWithTagCommand.java

```diff
@@ -14,88 +13,70 @@
 
     private final Set<Tag> targetTags;
 
     private final List<Person> deletedPersons;
+    private final boolean isExact;
 
     /**
      * Creates an DeleteContactWithTagCommand to delete the {@code Person} with specified {@code Tag}
      */
-    public DeleteContactWithTagCommand(Set<Tag> targetTags) {
+    public DeleteContactWithTagCommand(Set<Tag> targetTags, boolean isExact) {
         this.targetTags = targetTags;
         deletedPersons = new ArrayList<>();
+        this.isExact = isExact;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        List<Person> personList = model.getPersonListCopy();
+        List<Person> personList = model.getFilteredPersonList();
 
         for (Person person : personList) {
-            removePersonWithTags(model, person);
+            checkToBeDeleted(model, person);
+        }
+        for (Person person : deletedPersons) {
+            model.deletePerson(person);
         }
 
         // Only save state if there are changes (person deleted)
         if (!deletedPersons.isEmpty()) {
             model.addState();
-        }
-
-        return new CommandResult(String.format(MESSAGE_DELETE_TAGS_SUCCESS, displayTags())
-                + (deletedPersons.isEmpty()
-                ? ""
-                : String.format("\n" + MESSAGE_DELETE_PERSON_SUCCESS, displayPersons())));
-    }
-
-    private void removePersonWithTags(Model model, Person person) {
-        Set<Tag> tags = new HashSet<>(person.getTags());
-
-        boolean isUpdated = false;
-        for (Tag t : targetTags) {
-            if (tags.remove(t)) {
-                isUpdated = true;
-            }
-        }
-
-        // Dont delete people who not updated
-        if (!isUpdated) {
-            return;
-        }
 
-        if (tags.isEmpty()) {
-            deletedPersons.add(person);
-            model.deletePerson(person);
+            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, displayPersons()));
         } else {
-
-            Person editedPerson = new Person(person.getName(),
-                    person.getPhone(),
-                    person.getEmail(),
-                    person.getBirthday(),
-                    person.getAddress(),
-                    person.getRemark(),
-                    tags);
-
-            model.setPerson(person, editedPerson);
+            return new CommandResult(MESSAGE_PERSON_NOT_REMOVED);
         }
-
     }
 
     /**
-     * Returns tags in the form "a, b, c,..."
+     * Check if person tagged with the target tags
      */
-    private String displayTags() {
-        assert targetTags.size() > 0;
-        return targetTags.stream()
-                .map(t -> t.toString())
-                .map(s -> s.substring(1, s.length() - 1))
-                .reduce((a, b) -> a + ", " + b)
-                .get();
+    private void checkToBeDeleted(Model model, Person person) {
```

Comment:
Method name is a little misleading, as it seems to imply a check (i.e. returns a boolean). I think this method shouldn't be calling deletedPersons.add(person), as it might violate some abstraction principles.

It might be good to abstract this out into a `boolean hasTags()` method, that checks these things, and then leaving the adding to deletedPersons to execute.

Reply:
Yup makes sense!
Edited, info in comment below